src/hotspot/share/gc/shenandoah/shenandoahHeap.hpp

```diff
@@ -506,6 +506,11 @@ class ShenandoahHeap : public CollectedHeap {
   void sync_pinned_region_status();
   void assert_pinned_region_status() NOT_DEBUG_RETURN;
 
+// ---------- Concurrent Stack Processing support
+//
+public:
+  bool uses_stack_watermark_barrier() const { return true; }
```

Comment:
This overrides a CollectedHeap method. I wonder if we should start adding 'override' keywords to overridden methods. However, this requires >=C++11, and might disturb backporting. Not sure about it.